doc/Messaging.md

```diff
@@ -0,0 +1,219 @@
+ethnicity.Add(CodingResponseMessage.HispanicOrigin.DETHNICE, <ethnicity code>);
+ethnicity.Add(CodingResponseMessage.HispanicOrigin.DETHNIC5C, <ethnicity code>);
+message.Ethnicity = ethnicity;
+
+// Create the race coding
+var race = new Dictionary<CodingResponseMessage.RaceCode, string>();
+race.Add(CodingResponseMessage.RaceCode.RACE1E, <race code>);
+race.Add(CodingResponseMessage.RaceCode.RACE17C, <race code>);
+race.Add(CodingResponseMessage.RaceCode.RACEBRG, <race code>);
+message.Race = race;
+
+// Create the cause of death coding
+message.UnderlyingCauseOfDeath = <icd code>;
+
+// Assign the record axis codes
+var recordAxisCodes = new List<string>();
+recordAxisCodes.Add(<icd code>);
+recordAxisCodes.Add(<icd code>);
+recordAxisCodes.Add(<icd code>);
+recordAxisCodes.Add(<icd code>);
+message.CauseOfDeathRecordAxis = recordAxisCodes;
+
+// Assign the entity axis codes
+var builder = new CauseOfDeathEntityAxisBuilder();
+// for each entity axis codes
+...
+builder.Add(<lineNumber>, <positionInLine>, <icd code>);
+...
+// end loop
+message.CauseOfDeathEntityAxis = builder.ToCauseOfDeathEntityAxis();
+
+// Create a JSON representation of the coding response message
+string jsonMessage = message.ToJSON();
+
+// Send the JSON message
+...
+```
+
+On receipt of the `CodingResponseMessage`, the jurisdiction can parse it, determine the type of the message, and extract the required information using the following steps:
+
+```cs
+// Get the message as a stream
+StreamReader messageStream = ...;
+
+// Parse the message
+BaseMessage message = BaseMessage.Parse(messageStream);
+
+// Switch to determine the type of message
+switch(message)
+{
+    case CodingResponseMessage coding:
+        string nchsId = coding.NCHSIdentifier;
+        string stateId = coding.StateIdentifier;
+        string cod = coding.CauseOfDeathConditionId;
+        Dictionary<CodingResponseMessage.HispanicOrigin, string> ethnicity = coding.Ethnicity;
+        Dictionary<CodingResponseMessage.RaceCode, string> race = coding.Race;
+        List<string> recordAxis = coding.CauseOfDeathRecordAxis;
+        List<CauseOfDeathEntityAxisEntry> entityAxis = coding.CauseOfDeathEntityAxis;
+        ProcessCoding(nchsId, stateId, ethnicity, race, cod, recordAxis, entityAxis, );
+        break;
+    ...
+}
+```
+
+### Acknowledge Coding
+
+If extraction of the coding information was succesful, the jurisdiction can generate an acknowledgement message and send it to NCHS. As described earlier, the `AckMessage` constructor initializes properties based on the source coding message.
+
+The constructor will default the value of the `MessageSource` property to `http://nchs.cdc.gov/vrdr_submission` which is not suitable for messages originating from jurisdictions so this needs to be overriden.
+
+```cs
+// Create the acknowledgement message
+var ackMsg = new AckMessage(coding);
+ackMsg.MessageSource = "https://example.com/juristdiction/message/endpoint";
```

Comment:
Is setting MessageSource outside the constructor just to make it explicit to the reader?

Reply:
That was the idea. If we make the change above I'd remove this though.

Reply:
I think the last bit is tweaking this example to match the code changes and then we're good to merge